### PR TITLE
Cart Popup: remove CART_POPUP_OPEN Flux actions and "keep searching" logic

### PR DESCRIPTION
**Pull request by jsnajdr**
Removes some unused code I noticed when hacking on the shopping cart.

The `CART_POPUP_OPEN` and `CART_POPUP_CLOSE` Flux actions are no longer used anywhere. They used to be used (at least that's my guess) to show the shopping cart popup after selecting a plan, domain name or GSuite plan. The message to the user was like "I added the domain to the cart and I'm showing you. You can keep searching for another domain if you want to."

After these actions are removed from `PlansNavigation`, the "keep searching" logic that spans several components (`PopoverCart`, `CartButtons`, `GSuiteDialog`) is unused and can be removed.

Then the `CartButtons` component can be migrated from `React.createClass` to ES6 class, because the use of analytics mixin has been migrated to the Redux analytics uses everywhere else.

**How to review and test:**
This PR is best reviewed commit by commit.

Verify that the cart popover on the `/plans` page is shown correctly:
Screenshot 2019-05-15 at 11 53 01

### CODE REVIEW

**Review by klimeryk — APPROVED** [review]
Tested this to the best of my ability and everything looks good - thanks for cleaning up the stale code! 🙇

File: client/my-sites/checkout/cart/cart-buttons.jsx [hunk 1]
```diff
@@ -6,71 +6,47 @@
...
-import AnalyticsMixin from 'lib/mixins/analytics';
+import Button from 'components/button';
+import { recordGoogleEvent } from 'state/analytics/actions';
 
-export const CartButtons = createReactClass( {
-	displayName: 'CartButtons',
-	mixins: [ AnalyticsMixin( 'popupCart' ) ],
-
-	propTypes: {
+export class CartButtons extends React.Component {
```
Comment (flootr) [inline client/my-sites/checkout/cart/cart-buttons.jsx]: nit: there's only a single button in that component so I wonder whether we should rename this component to a singular case.

**Review by flootr — APPROVED** [review]
works well, thanks 🚢

File: client/my-sites/checkout/cart/cart-buttons.jsx (hunk 1 above)
Comment (jsnajdr) [reply client/my-sites/checkout/cart/cart-buttons.jsx]: The component is a button toolbar, and it happens to have just one button right now. I'd rather keep the name as it is. The possibility to add another button is still open.

Comment (flootr) [reply client/my-sites/checkout/cart/cart-buttons.jsx]: fine by me 👍

**Pull request merged**